cmd/ubuntu-image/main.go

```diff
@@ -191,6 +193,16 @@ func main() { //nolint: gocyclo
 		imageType = parser.Command.Active.Name
 	}
 
+	// Properly handle signals to execute defered functions and make sure
+	// mounted dir are unmounted
+	ch := make(chan os.Signal, 2)
+	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
```

Comment:
Do you already have plans for the signal handler? Like, if the cleanup process blocks or takes too long for some reason (like trying to call umount on a problematic mounting point (I don't know, imagine a bad disk or NFS mount that got stuck)) it should be possible to still interrupt the tool (you could kill -9 it anyway).

If you ctrl+c again while the signal handler is running nothing will happen because the signal is being caught. You could handle subsequent ctrl+c's and inform the user that the tool is shutting down but if they want to stop it anyway they can hit ctrl+c again.

Reply:
Yes! That is the plan in the following PR https://github.com/canonical/ubuntu-image/pull/235 (still very much WIP)